### DISCUSSION TITLE
Problem after login with azure

### DISCUSSION THREAD

**Discussion by ReneGraetz** [outline/outline#2089] [General] [answered]
Hello,

i followed these instructions here:
https://wiki.generaloutline.com/share/dfa77e56-d4d2-4b51-8ff8-84ea6608faa4

After Login this here happens:

ClientError [BadRequestError]: State not return in OAuth flow
    at new OAuthStateMismatchError (/opt/outline/build/server/errors.js:97:34)
    at StateStore.verify (/opt/outline/build/server/utils/passport.js:38:25)
    at Strategy.OAuth2Strategy.authenticate (/opt/outline/node_modules/passport-oauth2/lib/strategy.js:219:26)
    at Strategy.authenticate (/opt/outline/node_modules/@outlinewiki/passport-azure-ad-oauth2/lib/passport-azure-ad-oauth2/strategy.js:81:41)
    at attempt (/opt/outline/node_modules/passport/lib/middleware/authenticate.js:366:16)
    at authenticate (/opt/outline/node_modules/passport/lib/middleware/authenticate.js:367:7)
    at /opt/outline/node_modules/@outlinewiki/koa-passport/lib/framework/koa.js:194:7
    at new Promise (<anonymous>)
    at /opt/outline/node_modules/@outlinewiki/koa-passport/lib/framework/koa.js:193:12
    at /opt/outline/node_modules/@outlinewiki/koa-passport/lib/framework/koa.js:143:7
    at new Promise (<anonymous>)
    at passportAuthenticate (/opt/outline/node_modules/@outlinewiki/koa-passport/lib/framework/koa.js:107:15)
    at passportMiddleware (/opt/outline/build/server/middlewares/passport.js:53:7)
    at dispatch (/opt/outline/node_modules/koa-compose/index.js:44:32)
    at next (/opt/outline/node_modules/koa-compose/index.js:45:18)
    at /opt/outline/node_modules/koa-router/lib/router.js:321:16 {
  id: 'state_mismatch'

any ideas?

thanks a lot

Rene

**Comment by tommoor** [comment]
This would suggest that the `state` cookie sent in the response from `/auth/azure` was not set in the clients browser, perhaps a mismatching domain? You should look at the response payload from that endpoint in the browser developer tools for further clues

**Reply by ReneGraetz** [reply]
Thanks for your respond but i dont know how to do that.
Domain mismatch is possible.

I have the outline server
with this config

# URL should point to the fully qualified, publicly accessible URL. If using a
# proxy the port in URL and PORT may be different.
URL=https://External.url.de
PORT=443

Then there is a nginx reverse proxy in front of that. I can enter the site and its secured with lets encrypt.

On azure i configured the same domain.

What else can i forgot?

**Reply by ReneGraetz** [reply]
Ah one more thing.
when i check the last logins it says success and the correct app.

**Reply by tommoor** [reply]
It looks like this in dev tools, you must select "Preserve log" and then inspect the request to `/auth/slack` to see the cookie details:

![image](https://user-images.githubusercontent.com/380914/116335446-c1e16900-a78b-11eb-892b-db0144c7fe30.png)

You should see the state cookie being set with the correct domain. You can also access the cookies directly under the Application tab. If you don't know how to do this then it could be that cloud hosting would be a better solution for you

**Comment by ReneGraetz** [comment]
Ok think you are right. The response cookie is the internal ip.

![image](https://user-images.githubusercontent.com/78431484/116353285-c5f6a080-a7f6-11eb-925f-c3dff0e60d60.png)

**Comment by ReneGraetz** [comment, marked as answer]
Ok found some workarround for nginx:

server {
        server_name your.domain.de;
 location / {
                   proxy_cookie_domain localhost your.domain.de; <-- set here the login domin
                   proxy_pass http://172.X.X.X:443;
                    root /var/www/letsencrypt;
  }


    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/XXX/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/XXXX/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

**Reply by tommoor** [reply]
great, glad it's figured out!

**Comment by ReneGraetz** [comment]
Ah that was only working for a short time:

Now im getting:

2021/04/28 08:37:43 [error] 11208#0: *871 connect() failed (111: Connection refused) while connecting to upstream, client: 31.X.216.26, server: youd.domain.de, request: "GET /static/service-worker.js HTTP/1.1", upstream: "http://172.X.X.X:443/static/service-worker.js", host: "your.domain.de"

Is a reverse proxy not supported?

**Comment by ReneGraetz** [comment]
Ok foud a solution

server {
        server_name your.domain.de;


        add_header Strict-Transport-Security "max-age=31536000; preload";
        add_header HTTPS "on";

        if ($ssl_protocol = "") {
        return 301 https://$server_name$request_uri;
        }



 location / {
                proxy_set_header Host $host;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;proxy_set_header Host $host;
                   #proxy_cookie_domain localhost your.domain.de;
                   proxy_pass http://172.X.X.X:443;
                    root /var/www/letsencrypt;
  }


    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/your.domain.de/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/your.domain.de/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot
